UIforETW/ChildProcess.cpp

```diff
@@ -214,14 +219,14 @@ void ChildProcess::WaitForCompletion(bool printOutput)
 	if (hChildThread_)
 	{
 		WaitForSingleObject(hChildThread_, INFINITE);
-		CloseHandle(hChildThread_);
+		handle_close::closeHandle(hChildThread_);
 		hChildThread_ = 0;
 	}
 
 	// Clean up.
 	if (hPipe_ != INVALID_HANDLE_VALUE)
 	{
-		CloseHandle(hPipe_);
+		handle_close::closeHandle(hPipe_);

```

Comment:
For your convenience, here's the email, with the part about the annotation:

> [....]I came up with the idea of wrapping CloseHandle with a function like this:
```
_Success_( return )
BOOL safeCloseHandle( _Pre_valid_ _Post_ptr_invalid_ HANDLE handle_to_close )
{
	return CloseHandle( handle_to_close );
}
```
It was after I made the connection that a HANDLE is just a `typedef`d `void*`, and thus it is "invalidated" in a sense by calling CloseHandle. It works beautifully in contrived examples; if you try to call safeCloseHandle (or a better name`*`) twice, then it is reliably picked up by /analyze, with a nice "using uninitialized memory" error. I've been planning to try it on a large codebase since then, particularly Notepad++, but Notepad++ has a fairly slow development cycle. I've also been planning an article on it, so trying it on a large project would be a neat validation of the idea.

> So I decided a few minutes ago to try it on UIforETW, just to see how it'd work. There are only ~10 calls to CloseHandle, so I didn't expect any bugs (I didn't find any), but safeCloseHandle did work exactly as it did in smaller examples.

> If you open up [the toy-around commit that I pushed](https://github.com/ariccio/main/commit/8d9a91cfa4f211be61e225e7b435408308920f7a), uncomment any of the duplicated calls to safeCloseHandle, and rebuild, /analyze catches it. Comment out the duplicate safeCloseHandle calls, add some duplicate CloseHandle calls, and rebuild, no complaint from /analyze. I think it's pretty cool.

> `*safeCloseHandle` is probably a bad name, as it gives a "positive" illusion of safety, and thus people will abuse/misinterpret it. It should really be something like "checkedCloseHandle", or "checked_wrappers:close_handle".